src/rafter_consensus_fsm.erl

```diff
@@ -841,25 +823,39 @@ become_candidate(#state{term=CurrentTerm, me=Me}=State0) ->
     _ = request_votes(State3),
     State3.
 
-become_leader(#state{me=Me, term=Term, init_config=InitConfig}=State) ->
-    NewState = State#state{leader=Me,
+become_leader(#state{me=Me, term=Term, config=Config, init_config=InitConfig, timer=Timer}=State) ->
+    NewState0 = State#state{leader=Me,
                            responses=dict:new(),
                            followers=initialize_followers(State),
                            send_clock = 0,
                            send_clock_responses = dict:new(),
-                           read_reqs = orddict:new()},
+                           read_reqs = orddict:new(),
+                           timer = Timer},
 
     case InitConfig of
         complete ->
             %% Commit a noop entry to the log so we can move the commit index
             Entry = #rafter_entry{type=noop, term=Term, cmd=noop},
-            append(Entry, NewState);
-        _ ->
-            %% First entry must always be a config entry
-            NewState
+            append(Entry, NewState0);
+        undefined ->
```

Comment:
I don't believe that this is a valid state. Please remove the entire `undefined` clause. If for some reason we end up here the system should crash as this is intentionally undefined behavior. We should instead prevent followers from becoming candidates with an undefined initial_config, which I believe we already do.